modules/core/js_modules/route_handlers.js

```diff
@@ -93,6 +93,21 @@ function applySearchPageHandlers(routeParams) {
     $('.search_reset').on("click", Hm_Utils.reset_search_form);
     $('.combined_sort').on("change", function() { Hm_Message_List.sort($(this).val()); });
 
+    // Add explicit handler for search form submission to fix double-click issue
+    $('.search_update').on("click", function(e) {
```

Comment:
.search_update is a submit button. We shouldn't need to emulate its behaviour of form submission. The proper fix here would be to find the reason why clicking it one time doesn't submit the form and fix that - probably other event handlers kicking in?